File: Algorithm/src/edu/Q25501.java
package edu;

import java.io.BufferedReader;
import java.io.InputStreamReader;

public class Q25501 {
	
	public static int recursion(String s, int l, int r, int count){
        if(l >= r) {
        	System.out.println("1 " + count);
        	return 1;
        }
        else if(s.charAt(l) != s.charAt(r)) {
        	System.out.println("0 " + count);
        	return 0;
        }
        else return recursion(s, l+1, r-1, count + 1);
    }
    public static int isPalindrome(String s){
    	int count = 0;
        return recursion(s, 0, s.length()-1, count + 1);
    }

	public static void main(String[] args) throws Exception {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		int n = Integer.parseInt(br.readLine());
		
		for(int i = 0; i < n ; i++) {
			isPalindrome(br.readLine());
		}
	}

}

File: to python/Q25501.py
import sys

def recursion(s, l, r, count):
    if l >= r:
        print("1", end = " ")
        print(count)
        return 1
    elif s[l] != s[r]:
        print("0", end = " ")
        print(count)
        return 0
    else: return recursion(s, l+1, r-1, count + 1)

def isPalindrome(s):
    count = 0
    return recursion(s, 0, len(s)-1, count + 1)

if __name__ == "__main__":

    n = int(sys.stdin.readline())
    
    for i in range(n):
        isPalindrome(sys.stdin.readline())
